e2e/transaction-details.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { expect } from '@playwright/test'
+import { Transaction, User } from '@prisma/client'
+import { printRupiah } from '../app/utils/locales'
+import { TRANSACTION_STATUS } from '../app/models/enum'
+import { test } from './base-test'
+
+const validTransactionData: Transaction & { user: Partial<User> } = {
+  id: 'baaf91d1-cd03-498c-8b5d-fd46c123b686',
+  subscriptionId: 'a41ee28d-ca47-4a70-ab45-cf5d0b66734c',
+  userId: '2c15a165-68b2-4b07-8a2e-38aa6345f308',
+  bankName: 'Bank Mandiri',
+  bankAccountName: 'Pejuang Kode',
+  bankAccountNumber: '123456789',
+  amount: 25000,
+  status: TRANSACTION_STATUS.SUBMITTED,
+  createdAt: new Date(1641283151245),
+  updatedAt: new Date(1641283151245),
+  user: {
+    name: 'Pejuang Kode',
+    phoneNumber: '-',
+  },
+}
+
+const validTransactionData1 = {
+  id: '817cc1bb-f821-4fb4-b643-46c098919ce3',
+  user: {
+    phoneNumber: '628999210188',
+  },
+}
+
+test.use({
+  storageState: 'e2e/auth.json',
+})
```

Comment:
It is concerning that we have another testing fixture (other than the `auth.json` file) that we need to update whenever we reset the `test.db` using this command:

```sh
DATABASE_URL="file:./test.db" npx prisma migrate reset
```

For instance:

![Reset test.db using a new data seed](https://user-images.githubusercontent.com/6315466/148246054-e53344ec-b603-4601-b038-5b96f82a9359.png)

After running that command above, we will get these errors when we run this command:

```sh
npm run test:e2e:run
```

![Failing end-to-end tests due to wrong fixtures](https://user-images.githubusercontent.com/6315466/148246171-1e67eb5d-7715-4811-ac40-4b69e85c826e.png)

Let's try to address this concerning problem in another issue.

Reply:
Ah yeahi think it was my fault, because i hardcoded the transaction id (which will be different each time the seeds is run)  in the fixtures, maybe we can retrieve the transaction data from db instead of using fixtures